client/landing/stepper/declarative-flow/flows/onboarding-unified/onboarding-unified.ts

```diff
@@ -0,0 +1,157 @@
+import { Onboard, OnboardActions, UserSelect, Visibility } from '@automattic/data-stores';
+import { ONBOARDING_UNIFIED_FLOW } from '@automattic/onboarding';
+import { dispatch, useSelect } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { translate } from 'i18n-calypso';
+import { SIGNUP_DOMAIN_ORIGIN } from 'calypso/lib/analytics/signup';
+import {
+	clearSignupDestinationCookie,
+	setSignupCompleteFlowName,
+	persistSignupDestination,
+	setSignupCompleteSlug,
+} from 'calypso/signup/storageUtils';
+import { ONBOARD_STORE, USER_STORE } from '../../../stores';
+import { usePurchasePlanNotification } from '../../internals/hooks/use-purchase-plan-notification';
+import { STEPS } from '../../internals/steps';
+import { ProcessingResult } from '../../internals/steps-repository/processing-step/constants';
+import type { FlowV2, SubmitHandler, ProvidedDependencies } from '../../internals/types';
+
+function initialize() {
+	const { setHidePlansFeatureComparison, setIntent } = dispatch( ONBOARD_STORE ) as OnboardActions;
+
+	// Set up the flow defaults
+	setHidePlansFeatureComparison( false ); // Show plan comparisons for conversion
+	clearSignupDestinationCookie();
+	setIntent( Onboard.SiteIntent.Build ); // Default to build intent
+
+	// Minimal flow: plans -> siteless checkout -> post-checkout processing
+	// No login required - authentication will happen during checkout
+	// Need processing and post-checkout steps for Commerce plan atomic conversion
+	return [ STEPS.UNIFIED_PLANS, STEPS.POST_CHECKOUT_ONBOARDING, STEPS.PROCESSING, STEPS.ERROR ];
+}
+
+const onboardingUnifiedFlow: FlowV2< typeof initialize > = {
+	name: ONBOARDING_UNIFIED_FLOW,
+
+	isSignupFlow: true,
+	initialize,
+	useStepNavigation( _currentStep, navigate ) {
+		const flowName = this.name;
+		const userIsLoggedIn = useSelect(
+			( select ) => ( select( USER_STORE ) as UserSelect ).isCurrentUserLoggedIn(),
+			[]
+		);
+
+		const { setSignupDomainOrigin } = dispatch( ONBOARD_STORE ) as OnboardActions;
+		const { setShouldShowNotification } = usePurchasePlanNotification();
+
+		/**
+		 * Get post-checkout destination for onboarding-unified flow (simplified version)
+		 */
+		const getPostCheckoutDestination = async (
```

Comment:
This function doesn't seem async.

Reply:
Thanks for catching that!
I must of missed it after simplifying it.